### ISSUE TITLE
http://sai.unich.it/submit-match  404 error

### ISSUE THREAD

**Issue by Cabu** [CuriosAI/sai#102]
When the client try to access http://sai.unich.it/submit-match  it receive a 404. I don't think it is normal :(

**Comment by amato-gianluca** [comment]
It isn't, but is this still happening?

Note that going to http://sai.unich.it/submit-match with he web browser and
obtaining the 404 error IS normal.

--gianluca amato

**Comment by Cabu** [comment]
I have an error 22 for each match autogp try to send with curl. That's why i have tried to access the url with my browser...
![image](https://user-images.githubusercontent.com/453185/72684185-b0b4a180-3ade-11ea-8194-2319ae8607c7.png)

**Comment by amato-gianluca** [comment]
SAI is regularly receiving new games, therefore it must be something
related to your account. A couple of questions:

1) which version of SAI are you using?
2) does it used to work for you in the past?

--gianluca

**Comment by Cabu** [comment]
1) I use sai-0.17.5-gpu since I have started contributing around 10 days ago
2) It should have worked because It has played and registered 7714 games until now under the name Kabu

From curl documentation an error 22 CURLE_HTTP_RETURNED_ERROR (22) is when the HTTP server returns an error code that is >= 400. 

When I do a curl.exe http://sai.unich.it/get-task/0, I receive:
{"cmd":"selfplay","hash":"ae9ceeec4f141bf1dd077b42c849c9ef5bcd3705fc9d53940255000007ceee8e","required_client_version":"16","minimum_autogtp_version":"16","random_seed":"8352012539460419828","minimum_leelaz_version":"0.15","options_hash":"b9d280","options":{"playouts":"0","visits":"3500","resignation_percent":"5","noise":"true","randomcnt":"30","komi":"7","noise_value":"0.03","lambda":"0.1","other_options":"--nrsymm --restrict_tt --adv_features --chainlibs_feat --chainsize_feat --blunderthr 0.05 --randomtemp 1.0 --policy_temp 1.2 --recordvisits","dumbpass":"false"},"hash_gzip_hash":"d922a4a6ae603aa2d3a7bf4fdf1890df3ee70c5a856cbad7a2b36ec772c80440"}

But When I do it with my login and password (curl.exe http://sai.unich.it/get-task/0 -F username=Kabu -F password=xxxxxx) I receive:
Invalid authentication

**Comment by Vandertic** [comment]
You should pass the hashed password when using curl directly. Instead you should give the bare password when using autogtp.

This said, I don't know what your problem could be, apart from having a wrong password.

You can create another username with the same e-mail if you want, or maybe @amato-gianluca can reset your password in some way?

**Comment by Cabu** [comment]
I am sure the login/password are correct because they are stored in the batch file I used before making sai.hta work on my computer, the sai.cfg used by sai.hta and my password manager and they all match. 

But I have nonetheless created a new account and I still get the same error.

**Comment by Cabu** [comment]
I deleted the file curl_save5c32d27f869a4c53b0efd039ae50b11c.bin and everything is back to normal.

**Comment by amato-gianluca** [comment]
Good to know! Thanks for reporting your problem.

**Comment by Cabu** [comment]
I got the same problem again. Deleting the temporary curl file resolved it. I don't have a way to reproduce the problem :(

**Comment by nanzi** [comment]
I guess 10 days before you met this issue, you had saved a match game and quit.
10 days later, the match sgf file was always rejected by the server, since match game would be 60/50 full. The server strategy is this: if a client request a match, but could not upload in time, the server will rearrange a new client to do a match.
Sometimes I found a match game will be saved even I closed the console directly,not with a 'q' input.

**Comment by egri-nagy** [comment]
I have exactly the same problem now. I will check whether removing temporary files could help or not.

**Comment by egri-nagy** [comment]
Yes, removing the temporary files solves the problem, but I lost around 15 match games. I will watch closely how this happens.
